shell/spy.go

```diff
@@ -1,19 +1,32 @@
 package shell
 
 import (
-	"bytes"
+	"fmt"
+	"io"
 	"log"
 	"os/exec"
+	"strings"
 )
 
-// Execute accepts a command argument and returns the result of the executed command
-func Execute(script string) string {
-	cmd := exec.Command(script)
-	var out bytes.Buffer
-	cmd.Stdout = &out
+// ParseCommand accepts a string and returns the command name and args
+func ParseCommand(cmd string) (string, []string) {
+	stringResult := strings.Split(cmd, " ")
+
+	if len(stringResult) == 1 {
```

Comment:
Although `strings.Split` guarantees to return a slice of at least one element, we can be still more cautious here:

```suggestion
	if len(stringResult) <= 1 {
```

(And since `shell.Split` doesn't make any such guarantee, this would be wise.)

This is an example of what I call the 'cup of tea test'. If the reader got up to make a cup of tea between the call to `strings.Split` and this `if` statement, she might have forgotten that we can't have a zero-length slice, and be puzzled about why this is `==` instead of `<=`. In other words, it's possible to write completely correct code that nonetheless _looks_ wrong, and this creates a tiny speed bump for readers.